### DISCUSSION TITLE
Will rust-gpu support other binary shader targets?

### DISCUSSION THREAD

**Discussion by BarronKane** [EmbarkStudios/rust-gpu#1087] [General]
As it stands I see only support for SPIR-V compilation, which is fine on it's own! This project looks wonderful but I'm also wondering if there are plans (or incompatibilities) in the core libraries for compiling to another target like DXIL? 

Converting SPIR-V to DXIL can of course be done, but sometimes dubiously. Is rust-gpu being developed with future compatibility in mind? I.e. are descriptor sets using a flat binding scheme or is that up to the consuming crate? Etc.

Obviously SPIR-V and DXIL house very different and incompatible formats of IR, and there was some talk about DXIL joining the modern era of LLVM versions in the mainline, but it seams your team wrote a custom compiler for SPIR-V? If so that's kind of insane, and I was going to look to contributing for DXIL but I'm starting to see just how complex a problem that would be.

I'm just looking for your thoughts and outlook on the mid to long term goals on this project. I'm going to try using it on the RHI I'm building for my hobbiest renderer, and I think both inside AND outside of the Rust ecosystem the potential here is enormous. I'll certainly push up anything I find if an opportunity arises for me to contribute.

**Comment by repi** [comment]
No plans for targeting DXIL or other shader formats at this time.

In the long term that would be nice, but would be huge amount of work so out of scope. Would require someone else that has a need for it to build it and support it long term.
